src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java

```diff
@@ -164,7 +165,29 @@ public boolean isUpvoteButtonVisible() {
     return upvoteButton.isDisplayed();
   }
 
+  public PostEntity clickFollowFirstPost() {
+    PostEntity post = findNewestPost();
+    post.clickFollow();
+    sleepForTwoSeconds();
+    return post;
+  }
+
+  @SneakyThrows(RuntimeException.class)
+  private void sleepForTwoSeconds() {
+    try {
+      TimeUnit.SECONDS.sleep(2);
+    } catch (InterruptedException x) {
+      PageObjectLogging.logInfo("Sleep interrupted");
+      throw new RuntimeException(x);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS00112)